### DISCUSSION TITLE
Compilation problem: unable to link to libegspp library (cannot find -legspp)

### DISCUSSION THREAD

**Discussion by phantasuson** [nrc-cnrc/EGSnrc#736] [Questions] [answered]
Hi all, I'm new to EGS, could any one help me please It came out when I compiling tutor7pp for the first try. Thanks in advance.

```
D: \EGSnrc \EGSnrc \EGSnrc \egs_ home \myapp>mingw32- make
g++ -I. -ID: \EGSnrc \EGSnrc \EGSnrc \HEN HOUSE\lib\win3264 ID: \EGSnrc \EGSnrc \EGSnrc \HEN HOUSE\egs++ ID: \EGSnrc \EGSnrc \EGSnrc \HEN_ HOUSE\interface -DWIN32 fPIC -02 mtune-native 一c -0 myapp_ win3264. 0 myapp. cpp
g++ -I. - ID: \EGSnrc \EGSnrc \EGSnrc \HEN_ HOUSE\lib \win3264 - ID: \EGSnrc EGSnrc \EGSnrc \HEN_ HOUSE\egs++ -ID: \EGSnrc \EGSnrc \EGSnrc HEN_ HOUSE \interface DWIN32 -fPIC -02 -mtune=native -C -0 egs_ advanced application win3264.0 D: \EGSnrc \EGSnrc \EGSnrc \HEN HOUSE \egs++\egs_ _advanced application. cpp
” Mortran compiling EGSnrc sources ..."
gfortran -02 -mtune=native fPIC    -C -0 egsnrc win3264. 0 egsnrc_ win3264. F
g++ -I. - ID: \EGSnrc \EGSnrc \EGSnrc \HEN HOUSE\1ib\win3264 -ID: \EGSnrc EGSnrc \EGSnrc \HEN HOUSE\egs++ -ID: \EGSnrc \EGSnrc EGSrc \HEN_ HOUSE\interface -DWIN32 -fPIC -02 -mtune=native -c -0 egs_ interface2 win3264.。D:\EGSnrc EGSnrc \EGSnrc \HEN HOUSE\interface\egs_ interface2. C
g++ -02 -mtune-native -0 D: \EGSnrc EGSnrc EGSnrc \egs_ home \bin win3264 \myapp. exe myapp_ win3264. 0 egs_ _advanced application_win3264. 0 egsnrc_ win3264.o egs_ _interface2 win3264. 0 -LD: \EGSnrc \EGSnrc \EGSnrc HEN_ HOUSE \egs++\dso\win3264 legspp -1gfortran
**d:/egsnrc/ mingw/bin/.. /lib/ gcc/mingw32/6. 3.0/../../.. /.. /mingw32/bin/ ld. exe: cannot find -legspp**
collect2. exe: error: 1d returned 1 exit status
D: EGSnre \EGSnrc \EGSnrc HEN_ HOUSE \makefiles \cpp_ makefile:44: recipe for target' D: \EGSnrc EGSnrc EGSnrc \egs_ home \bin \win3264 \myapp. exe’ failed
mingw32- make: *** [D: \EGSnrc EGSnrc EGSnrc \egs_ home \bin \win3264 \myapp. exe] Error 1 
D: \EGSnrc \EGSnrc \EGSnrc \egs_ home \myapp
```

**Comment by ftessier** [comment]
Hi @phantasuson, I converted this issue to a *Discussion*, which is our preferred forum for questions. The error you are reporting indicates that while compiling `myapp` the linker is not able to find the `libegspp` library, which is normally compiled as part of the EGSnrc configuration. Can you quickly retrace how you installed and configured EGSnrc, and if possible share the file `HEN_HOUSE/log/configure.log` that was created during your installation?

**Reply by phantasuson** [reply]
Thank you for your answer. I am on a business trip for three days. I will upload the log as soon as I come back

**Reply by phantasuson** [reply]
[config_win3264.log](https://github.com/nrc-cnrc/EGSnrc/files/6683304/config_win3264.log)
I followed the guide on GitHub to install it, but the package was downloaded from somewhere else

**Reply by phantasuson** [reply]
I reinstalled EGS and this time no errors were reported when compiling, but a new problem occurred at run time

```
D:\EGSnrc\EGSnrc\egs_home\myapp>myapp -i slab.egsinp
In EGS_Application::initSimulation()
EGS_Library::load(): failed to load library D:\EGSnrc\EGSnrc\HEN_HOUSE\egs++\dso\win3264\egs_ndgeometry.dll
  error was: 126
createSingleGeometry: Failed to load library 'egs_ndgeometry' from D:\EGSnrc\EGSnrc\HEN_HOUSE\egs++\dso\win3264
EGS_BaseGeometry::createGeometry: errors during geometry definition
```

**Comment by phantasuson** [comment]
I tried to reinstall it a few more times and checked the log, and it seemed like every compilation failure were caused by the same files.
[config_win3264.log](https://github.com/nrc-cnrc/EGSnrc/files/6685700/config_win3264.log)

**Reply by rtownson** [reply]
I have seen this error before when there was insufficient RAM - how much system memory do you have? You need at least 6GB to both run Windows and install EGSnrc.

**Reply by phantasuson** [reply]
I have 8GB in total, but I can only use 3GB or so. Maybe that's the reason

**Comment by phantasuson** [comment, marked as answer]
I expanded the RAM to 16G and successfully installed it, thanks for your help
